### PR TITLE
Ft setup signup and login endpoint 157214898

### PR DESCRIPTION
**Pull request by samuelbwambale**
Added JWT authentication and tests for signup/login are passing.
PT story: https://www.pivotaltracker.com/n/projects/2166953/stories/156978726

### CODE REVIEW

File: app/api/users_resource.py [hunk 1]
```diff
@@ -1,36 +1,78 @@
...
+                    help='Username is required')
+parser.add_argument('password', type=str, required=True,
+                    help='Password is required')
+parser.add_argument('isAdmin', type=bool, required=True,
+                    help='Status is required')
 
 
 class UsersResource(Resource):
     def get(self):
+        # return {
```
Comment (malmike) [inline app/api/users_resource.py:17]: Remove commented out code?

File: app/api/users_resource.py [hunk 2]
```diff
@@ -1,36 +1,78 @@
...
-        username = params['username']
-        password = params['password']
-        isAdmin = params['isAdmin']
-
-        user = User(username=roberts, password=pwd123, isAdmin=False)
-        user.addUser()
-        return jsonify ({ 'user': user  }), 200
-
+        data = parser.parse_args()
+        user = get_user_by_username(data['username'])
```
Comment (malmike) [inline app/api/users_resource.py:27]: Validate the data that you receive

File: app/api/users_resource.py [hunk 3]
```diff
@@ -1,36 +1,78 @@
...
+
+        if user:
+            return {'message': 'User {} already exists'.format(data['username'])}
```
Comment (malmike) [inline app/api/users_resource.py:30]: Add status codes

File: app/api/users_resource.py [hunk 4]
```diff
@@ -1,36 +1,78 @@
...
+            user.addUser()
+            access_token = create_access_token(identity = data['username'])
+            refresh_token = create_refresh_token(identity = data['username'])
+            return { 
+                'user': user.to_dict(),
+                'message': 'User {} was created'.format(data['username']),
+                'access_token': access_token,
+                'refresh_token': refresh_token
+                }, 201
+        except Exception as err:
```
Comment (malmike) [inline app/api/users_resource.py:47]: Try to get conditions under which certain issues occur like user already exists. Do not just generalise all the issues

**Comment by samuelbwambale** [comment]
Comments noted. Adjustments in progress

**Pull request merged**